internal/tenant/context.go

```diff
@@ -55,3 +55,12 @@ func TestContext() context.Context {
 	}
 	return withTenant(context.Background(), 1)
 }
+
+// TestContextWithID is like TestContext but allows you to specify the tenant
+// ID.
+func TestContextWithID(i int) context.Context {
+	if !testutil.IsTest {
+		panic("only call this function in tests")
+	}
+	return withTenant(context.Background(), i)
```

Comment:
idea: Could we say this creates a new tenantCtx but you don't get to control the ID?
```suggestion
// TestContextWithID is like TestContext but allows you to specify the tenant
// ID.
func NewTestTenantContext() context.Context {
	if !testutil.IsTest {
		panic("only call this function in tests")
	}
	return withTenant(context.Background(), nextTestTenant.Add(1))
```
? 

My thinking: This encourages to think less about the ID as being something you should care about as someone who writes tests; like in application code where you should also not care.

If you want to write a test that uses two tenants, you can simply call this twice and manage the two contexts yourself

Reply:
Sounds good to me!